cgen error for `Sumtype(AliasOfTheSumtype(Sumtype(...)))`

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.12 78f3cf3, press to see full `v doctor` output</summary>

|V full version      |V 0.4.12 fa49b769e15794f8cda962fd714c9c4292c1c99c.78f3cf3
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 20.04.6 LTS
|Processor           |4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
|Memory              |1.83GB/15.05GB
|                    |
|V executable        |/home/delian/v/v
|V last modified time|2025-10-01 05:27:42
|                    |
|V home dir          |OK, value: /home/delian/v
|VMODULES            |OK, value: /home/delian/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/delian/v
|                    |
|env VFLAGS          |"-no-parallel"
|Git version         |git version 2.50.0
|V git status        |weekly.2025.40-13-g9308ec1e
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
|gcc version         |gcc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
|clang version       |N/A
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (Ubuntu GLIBC 2.31-0ubuntu9.18) 2.31

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  x.v && /home/delian/v/x`
```v
struct Number {}

struct String {}

struct Variable {}

struct FunctionCall {}

type Value = Number | String | Variable | FunctionCall
type Statement = Value

s := Value(Statement(Value(Number{})))
println(s)

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/x.01K6JJJ23YFK70R1VESTAZENY6.tmp.c:1367: error: field not found: _main__Value
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
```

**What did you expect to see?**

a compiled program


> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -5595,6 +5595,10 @@ fn (mut g Gen) cast_expr(node ast.CastExpr) {
 	}
 	node_typ_is_option := node.typ.has_flag(.option)
 	if sym.kind in [.sum_type, .interface] {
+		if g.table.unaliased_type(expr_type) == node_typ {
+			g.expr(node.expr)
+			return
+		}
 		if node_typ_is_option && node.expr is ast.None {
 			g.gen_option_error(node.typ, node.expr)
 		} else if node.expr is ast.Ident && g.comptime.is_comptime_variant_var(node.expr) {

--- a/vlib/v/tests/sumtypes/sumtype_cast_alias_test.v
+++ b/vlib/v/tests/sumtypes/sumtype_cast_alias_test.v
@@ -1,0 +1,18 @@
+struct Number {}
+
+struct String {}
+
+struct Variable {}
+
+struct FunctionCall {}
+
+type Value = Number | String | Variable | FunctionCall
+type Statement = Value
+
+fn test_main() {
+	s := Value(Statement(Value(Number{})))
+	assert '${s}' == 'Value(Number{})'
+
+	s2 := Statement(Value(Number{}))
+	assert '${s2}' == 'Statement(Value(Number{}))'
+}

